source/contributed/advanced_grunt.md

```diff
@@ -0,0 +1,457 @@
+title: Advanced Grunt Usage
+---
+
+Uploading your code with grunt is just the beginning. This guide covers numerous enhancements that can be added to your scripts to really get the most out of grunt and make your development easier.
+
+
+## Assumptions
+
+This article makes a few assumptions-
+
+* Code is stored in the `src` directory.
+* You've read the [Getting Started](http://gruntjs.com/getting-started) guide.
+* Specifically, you understand how tasks are stored using grunt.initConfig.
+
+
+## Override Options with CLI Flags
+
+Changing the options that Grunt is using should not require a change in code and can be done using command line flags.
+
+First create the file `.screeps.json` to save the default options and your credentials.
+
+    {
+      "email": "<YOUR EMAIL HERE>",
+      "password": "<YOUR PASSWORD HERE>",
+      "branch": "default",
+      "ptr": false
+    }
+
+Now update `Gruntfile.js` so it utilizes the file and the `grunt.option` function.
+
+    module.exports = function(grunt) {
+
+        var config = require('./.screeps.json')
+        var branch = grunt.option('branch') || config.branch;
+        var email = grunt.option('email') || config.email;
+        var password = grunt.option('password') || config.password;
+        var ptr = grunt.option('ptr') ? true : config.ptr
+
+        grunt.loadNpmTasks('grunt-screeps');
+
+        grunt.initConfig({
+            screeps: {
+                options: {
+                    email: email,
+                    password: password,
+                    branch: branch,
+                    ptr: ptr
+                },
+                dist: {
+                    src: ['src/*.js']
+                }
+            }
+        });
+    }
+
+Now you can override any of the commands on the fly.
+
+    grunt screeps --ptr=true --branch=development
+
+
+## Using Folders
+
+A common frustration amongst new players is that folders are not available by default. This can be changed using Grunt.
+
+To get started install the "grunt-contrib-copy" and "grunt-contrib-clean" plugins.
+
+    npm install grunt-contrib-clean --save-dev
+    npm install grunt-contrib-copy --save-dev
+
+
+In this case the "copy" plugin is going to be used to move code from the `src` directory to `dist`. The plugin as an option to rename files, so a function to convert directory delimiters (slashes) to underscores is used to flatten the file structure.
```

Comment:
Would be nice to see here an example of modules and directory structure "before" and "after".